04_pig_latin/pig_latin.rb

```diff
@@ -0,0 +1,35 @@
+def logic(str)
+  vowels = %w[a e i o u]
+
+  until vowels.include?(str[0])
+    str = if str[0..1] == 'qu' || str[0..1] == 'Qu'
+            str[2..] << 'qu'
+          else
+            str[1..] << str[0]
+          end
+  end
+
+  str << 'ay'
+end
+
+def translate(str)
+  str.split.map { |item| logic(item) }.join(' ')
+end
+
+def capitalize(str)
+  str.split.map { |item|
+    item == item.capitalize ? translate(item).capitalize : translate(item)
+  }.join(' ')
```

Comment:
Still ain't a one-liner, but that's close enough.

Note than the ruby style is to use `{ }` notation when the code is on the same line, e.g.
```ruby
'a b c'.split.map { |letter| letter.upcase }
```
while for multi-line we use `do ... end` notation:
```ruby
'a b c'.split.map do |letter|
  letter
   .upcase
   .downcase
   .whatever
end
```